tables/tests/test_carray.py

```diff
@@ -2209,7 +2209,7 @@ def test00b_zeros(self):
             print("First row-->", ca[0])
             print("Defaults-->", ca.atom.dflt)
         self.assertTrue(allequal(ca[0], numpy.zeros(N, 'S3')))
-        self.assertTrue(allequal(ca.atom.dflt, numpy.zeros(N, 'S3')))
+        self.assertTrue(allequal(ca.atom.dflt, b""))
```

Comment:
Ok, so `'(3,)i4'` and `3*'i4'` are definitely different datatypes, the question is if we expect the promotion (demotion?) to a int32 array with `shape + (3,)` or not.

https://docs.scipy.org/doc/numpy/user/basics.rec.html#structured-arrays

Sorry for the stream of conscious comments as I sort this out :sheep: